src/utils/api-server.tsx

```diff
@@ -0,0 +1,460 @@
+          });
+        } else {
+          alert('login cancelled, you wont get jolokia access!');
+          console.log('cancelled');
+        }
+        return;
+      }
+
+      console.log('starting querying', testUrl);
+      if (testUrl == '') {
+        alert('you need to give a jolokia request url');
+        return;
+      }
+
+      const encodedUrl = testUrl.replace(/,/g, '%2C');
+      console.log('after encoded', encodedUrl);
+      fetch(encodedUrl, {
+        method: 'GET', // *GET, POST, PUT, DELETE, etc.
+        headers: {
+          'jolokia-session-id': getAuthToken(broker),
+        },
+      })
+        .then((result): void => {
+          if (result.ok) {
+            result.json().then((data) => {
+              console.log('get back result', data, 'utl', testUrl);
+              setJolokiaTestResult(JSON.stringify(data, null, 2));
+            });
+          } else {
+            result.text().then((data) => {
+              setJolokiaTestResult(data);
+            });
+          }
+        })
+        .catch((err) => {
+          setJolokiaTestResult(err);
+        });
+    } catch (error) {
+      setJolokiaTestResult(error);
+    }
+  };
+
+  const getBrokerRoute = (b: K8sResourceKind): K8sResourceKind => {
+    if (brokerRoutes.length > 0) {
+      const filteredRoutes = brokerRoutes.filter((route) =>
+        route.metadata.ownerReferences?.some(
+          (ref) =>
+            ref.name === b.metadata.name && ref.kind === 'ActiveMQArtemis',
+        ),
+      );
+      return filteredRoutes.length > 0 ? filteredRoutes[0] : null;
+    }
+    console.log('no route created');
+    return null;
+  };
+
+  return (
+    <>
+      <TextInput
+        value={testUrl}
+        type="text"
+        onChange={(value, _event) => setTestUrl(value)}
+        aria-label="text input example"
+        placeholder={
+          'input jolokia url here, example: https://' +
+          getApiHost() +
+          ':' +
+          getApiPort() +
+          getProxyUrl() +
+          '/api/v1/brokers'
+        }
+      />
+
+      <div>
+        <Button name="query" onClick={onButtonClick}>
+          Submit query
+        </Button>
+        &nbsp;&nbsp;&nbsp;
+        <Button name="info" onClick={onButtonClickShowApi}>
+          Show api info
+        </Button>
+        &nbsp;&nbsp;&nbsp;
+        <Button name="exec" onClick={onButtonClickExec}>
+          Test Operation
+        </Button>
+      </div>
+      <TextContent>
+        <TextArea
+          aria-label="auto resize"
+          autoResize={true}
+          value={jolokiaTestResult}
+          resizeOrientation="vertical"
+          rows={20}
+        ></TextArea>
+      </TextContent>
+    </>
+  );
+};
+
+interface LoginProp {
+  brokerDetail: K8sResourceKind;
+}
+
+export const LoginHandler: FC<LoginProp> = ({ brokerDetail }) => {
+  const brokerRoutes = useContext(RouteContext);
+
+  //none | ongoing | fail | ok
+  const [loginState, setLoginState] = useState('none');
```

Comment:
It's possible to define as a type the set of valid states: `const [loginState, setLoginState] = useState<'none' | 'ongoing' | 'fail' | 'ok'>('none');`